ENIGMAsystem/SHELL/Universal_System/Resources/fontstruct.cpp

```diff
@@ -36,35 +36,23 @@ using std::string;
 
 namespace enigma
 {
-  font **fontstructarray = NULL;
-  extern size_t font_idmax;
+  AssetArray<font, -1> fonts;
 
   bool fontglyph::empty() {
     return !(std::abs(x2-x) > 0 && std::abs(y2-y) > 0);
   }
 
   int font_new(uint32_t gs, uint32_t gc) // Creates a new font, allocating 'gc' glyphs
   {
-    font *ret = new font;
+    font ret;
 
     fontglyphrange fgr;
     fgr.glyphstart = gs;
-    fgr.glyphcount = gc;
```

Comment:
Okay, it's good that you got rid of this information after it was made redundant... but it shouldn't be redundant. This PR can ship with this as-is, but the right way to go about this is very likely to allocate the entire font as one block of memory that points into itself.